securedrop_client/gui/widgets.py

```diff
@@ -1807,6 +1803,208 @@ def _on_left_click(self):
             self.controller.on_submission_download(File, self.file.uuid)
 
 
+class ExportDialog(QDialog):
+
+    CSS = '''
+    #export_dialog {
+        min-width: 830;
+        min-height: 330;
+        border: 1px solid #2a319d;
+    }
+    '''
+
+    CSS_FOR_DIALOG_WITH_ERROR = '''
+    #export_dialog {
+        min-width: 830;
+        min-height: 430;
+        border: 1px solid #2a319d;
+    }
+    '''
+
+    CSS = '''
+    #export_dialog {
+        min-width: 400;
+        max-width: 400;
+        min-height: 200;
+        max-height: 200;
+    }
+    #passphrase_label {
+        font-family: 'Montserrat';
+        font-weight: 500;
+        font-size: 13px;
+    }
+    #passphrase_form QLineEdit {
+        border-radius: 0px;
+        min-height: 30px;
+        margin: 0px 0px 10px 0px;
+    }
+    '''
+
+    def __init__(self, controller, file_uuid):
+        super().__init__()
+
+        self.controller = controller
+        self.file_uuid = file_uuid
+
+        self.setObjectName('export_dialog')
+        self.setStyleSheet(self.CSS)
+        self.setWindowFlags(Qt.Popup)
+
+        layout = QVBoxLayout(self)
+        self.setLayout(layout)
+
+        # Starting export message
+        self.starting_export_message = SecureQLabel(_('Preparing export...'))
+        self.starting_export_message.setWordWrap(True)
+
+        # Widget to show error messages that occur during an export
+        self.generic_error = QWidget()
+        self.generic_error.setObjectName('gener_error')
+        generic_error_layout = QHBoxLayout()
+        self.generic_error.setLayout(generic_error_layout)
+        self.error_status_code = SecureQLabel()
+        generic_error_message = SecureQLabel(_('See your administrator for help.'))
+        generic_error_message.setWordWrap(True)
+        generic_error_layout.addWidget(self.error_status_code)
+        generic_error_layout.addWidget(generic_error_message)
+
+        # Insert USB Device Form
+        self.insert_usb_form = QWidget()
+        self.insert_usb_form.setObjectName('insert_usb_form')
+        usb_form_layout = QVBoxLayout()
+        self.insert_usb_form.setLayout(usb_form_layout)
+        self.usb_error_message = SecureQLabel(_(
+            'Either the drive is not encrypted with VeraCrypt, or there is something else wrong'
```

Comment:
Nit: right now we don't support veracrypt, we support luks. Even though finalizing wording is proposed in scope of #549 , it might be worth rewording this here to avoid confusion (and change to veracrypt once that is supported)